src/targets/services/helpers/model.js

```diff
@@ -10,12 +10,14 @@ export class Model {
   constructor() {
     this.uniqueY = Object.keys(classes)
     this.priors = Array(classes.length).fill(1)
-    this.occurences = Array(vocabulary.length).fill(
-      Array(this.uniqueY.length).fill(0)
-    )
-    this.logProbabilities = Array(vocabulary.length).fill(
-      Array(this.uniqueY.length).fill(0)
-    )
+
+    // Using map to allocate a new array for each line
+    this.occurences = Array(vocabulary.length)
+      .fill(0)
+      .map(() => Array(this.uniqueY.length).fill(0))
+    this.logProbabilities = Array(vocabulary.length)
+      .fill(0)
+      .map(() => Array(this.uniqueY.length).fill(0))
```

Comment:
Not sure why this change, is it more efficient?

Reply:
It was filling the matrices with the same array reference, as fill just takes a value (here, the reference of the array) and copies it for each element